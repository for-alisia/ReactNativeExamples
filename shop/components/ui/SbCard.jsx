// Dependencies
import React from 'react';
import { StyleSheet, View } from 'react-native';

// Theme
import theme from '../../theme';

const SbCard = (props) => {
  const { style, children } = props;

  return (
    <View style={{ ...styles.container, ...style }} testID="ui-card">
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
    shadowColor: 'black',
    shadowOpacity: 0.16,
    shadowOffset: { width: 0, height: 2 },
    borderRadius: theme.borderRadius.xs,
    elevation: 3,
    backgroundColor: theme.colors.white,
    flex: 1,
  },
});

export default SbCard;
