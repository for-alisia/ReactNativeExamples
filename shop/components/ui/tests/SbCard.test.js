// @ts-nocheck
// Dependencies
import React from 'react';
import { Text } from 'react-native';
import { render } from '@testing-library/react-native';

// Component
import SbCard from '../SbCard';

describe('SbCard UI Component', () => {
  const testText = 'Hello';
  const testElement = <Text>{testText}</Text>;

  it('renders a card with a child', () => {
    const { getByTestId, toJSON, getByText } = render(<SbCard>{testElement}</SbCard>);
    const card = getByTestId('ui-card');
    const text = getByText(testText);

    expect(card).toBeTruthy();
    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
