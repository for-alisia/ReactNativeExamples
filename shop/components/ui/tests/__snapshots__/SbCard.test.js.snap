// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbCard UI Component renders a card with a child 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "elevation": 3,
      "flex": 1,
      "overflow": "hidden",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.16,
    }
  }
  testID="ui-card"
>
  <Text>
    Hello
  </Text>
</View>
`;
